addons/mass_mailing/static/src/js/mass_mailing_html_field.js

```diff
@@ -78,7 +78,15 @@ export class MassMailingHtmlField extends HtmlField {
         popover.style.left = leftPosition + 'px';
     }
 
-    async commitChanges() {
+    async commitChanges(action) {
+
+        // When discarding changes, calling 'mass_mailing.commitChanges' will effectively make the form dirty,
+        // resulting in the reappearance of the discard button. To correct the flow, we first call 'super.commit'
+        // and then commit changes from 'mass_mailing' to complete the process.
+        if (action === "Discard changes") {
+            await super.commitChanges();
+        }
```

Comment:
also since we are overriding html Fields, commitChanges of parent should be triggered to save all changes IMO, try this remove if conditon and call super.commitChnages.? give it a try.